.travis.yml

```diff
@@ -0,0 +1,31 @@
+sudo: required
+
+language: scala
+
+services:
+- docker
+
+cache:
+  directories:
+  - $HOME/.ivy2/cache
+  - $HOME/.sbt/boot/
+
+script:
+  - /bin/true
```

Comment:
I kinda view the assembly JAR creation process as a step beyond compiling the source and running tests because it combines all of the dependent JARs into one big JAR. I moved it to the `before_deploy` block so that it would only trigger during a build that is marked for deployment.

It may be better to do something in `script` that forces a compilation. The noop for actually testing the code is not great. If we have something that can be used to test, exercising it in the `script` section would be nice.